src/models/differentiable/differentiable.jl

```diff
@@ -9,6 +9,12 @@ Base type for differentiable models written in pure Julia.
 """
 abstract type AbstractDifferentiableJuliaModel <: AbstractDifferentiableModel end
 
+"""
+Base type for models from the MLJ library.
+"""
+abstract type AbstractMLJModel <: AbstractDifferentiableModel end
+
```

Comment:
Is there actually a need to construct this struct? You created it only for the EvoTree model so far (which is differentiable afaik), so why don't just create this as a child class of an `AbstractDifferentiableModel`? I feel like we would need to create another struct for Flux models if we want to divide it that way.

Reply:
I think it's still worth having this struct, as either us or the client will add more MLJ models in the future. When adding generator support for these models, an abstract class will be needed for them, as the gradient information of MLJ models will be retrieved differently from the way the methods currently retrieve gradient information of the Flux models. However, I can remove this struct for now and it can be added once the package supports more MLJ models, if you think that's the cleaner solution.

Reply:
I don't know much about `LaplaceReduxModel` but maybe something like `CustomDifferentiableModel`? It'd be cool to have a different parent class for each package we're using models from + one additional class for our own implemented models. At the end of the day, we're using those parents' class only for the purpose of building a clear hierarchy, there's no functionality behind it.

Reply:
I changed it, let me know whether it looks fine to you now.